server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -536,8 +570,21 @@ public ReplicationStats execute(Client client) throws Exception {
 
       // If we have some edits to send
       if (0 < edits.walEdits.getEditsSize()) {
+        // Check if we are interrupted before to writing the edits
+        if (Thread.interrupted()) {
+          log.debug("Replication work interrupted before writing edits, returning empty replication stats");
+          return new ReplicationStats(0L, 0L, 0L);
+        }
+
         log.debug("Sending {} edits", edits.walEdits.getEditsSize());
         long entriesReplicated = client.replicateLog(remoteTableId, edits.walEdits, tcreds);
```

Comment:
> May still need some overall logic. If there are multiple client/RPC calls, the sum of them could exceed the replication timeout. For example if there are three RPC calls and each takes 9 mins and the replication timeout is 10 min, then it could run for 27 min without timing out.

I don't think that's the scope of what Adam is trying to fix. He's trying to set, for a single WAL being replicated, is the timeout for that replication task. So, for the number of threads that a TabletServer will devote to replicating, each of them would have the same amount of time to try to execute (10mins).